rcl/test/rcl/test_init.cpp

```diff
@@ -316,6 +319,76 @@ TEST_F(CLASSNAME(TestRCLFixture, RMW_IMPLEMENTATION), test_rcl_init_and_shutdown
   context = rcl_get_zero_initialized_context();
 }
 
+/* Tests rcl_init() deals with internal errors correctly.
+ */
+TEST_F(CLASSNAME(TestRCLFixture, RMW_IMPLEMENTATION), test_rcl_init_internal_error) {
+  rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+  rcl_ret_t ret = rcl_init_options_init(&init_options, rcl_get_default_allocator());
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_init_options_fini(&init_options)) << rcl_get_error_string().str;
+  });
+  FakeTestArgv test_args;
+  rcl_context_t context = rcl_get_zero_initialized_context();
+
+  {
+    auto mock = mocking_utils::patch_to_fail(
+      "lib:rcl", rmw_init, "internal error", RMW_RET_ERROR);
+    ret = rcl_init(test_args.argc, test_args.argv, &init_options, &context);
+    EXPECT_EQ(RCL_RET_ERROR, ret);
+    EXPECT_TRUE(rcl_error_is_set());
+    rcl_reset_error();
+    EXPECT_FALSE(rcl_context_is_valid(&context));
+  }
+
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    ret = rcl_init(test_args.argc, test_args.argv, &init_options, &context);
+
+    int64_t count = rcutils_fault_injection_get_count();
+    rcutils_fault_injection_set_count(RCUTILS_FAULT_INJECTION_NEVER_FAIL);
```

Comment:
May I ask, why setting this variable here? AFAIK thought the purpose of the `RCUTILS_FAULT_INJECTION_TEST` is to change that variable inside the fault injection test

Reply:
This is the current way of temporarily disabling fault injection. I don't want faults to be injected anywhere but within `rcl_init`.

Reply:
Yeah, it was clearer to me in the other test